### PR TITLE
chore: Properties on documentation pages

### LINKED ISSUE #238: Update documentation pages - component properties
On each component's page, the properties (shown as `@Inputs`) should be better defined including showing details for any sub-components.  For example, on the "Action Bar" page, there should be documentation for `ActionBar`, `ActionBarBack`, `ActionBarHeader` and `ActionBarActions`.

Refactor playground components to render required props and default values better.

Consider renaming `@Inputs` to `Properties`.

### PR DESCRIPTION
**Pull request by greg-a-smith**
### Description
I know this is a large PR, but since the changes to `Properties` were breaking (internally), all consuming components needed to be updated.  The `Properties` component will now use the source module to get all exported component names, then look up all props for each component based on `propTypes` and assemble the "Properties" section of the component documentation pages.

It gets "Name", "Type" and "Required" from `propTypes` and "Default" from `defaultProps`.  I've introduced a new static object for each component (where necessary) called `propDescriptions` that will populate the "Description".  There is a `defaults.js` file with property descriptions for some commonly used props and these will be used unless overridden.

I've also introduced support for Markdown in the property descriptions and the `Description` component.

#### Before:
screen shot 2019-01-24 at 11 40 21 pm

#### After:
screen shot 2019-01-24 at 11 39 38 pm

Fixes #238

### CODE REVIEW

File: src/Calendar/Calendar.js [hunk 1]
```diff
@@ -561,18 +561,39 @@ export class Calendar extends Component {
 
 }
 
-Calendar.propTypes = {
+Calendar.basePropTypes = {
```
Comment (greg-a-smith) [inline src/Calendar/Calendar.js]: Split these out so they could be shared by `DatePicker`.

File: src/Time/Time.js [hunk 2]
```diff
@@ -474,3 +437,69 @@ export class Time extends Component {
...
 }
+
+Time.basePropTypes = {
+    format12Hours: PropTypes.bool,
+    showHour: PropTypes.bool,
+    showMinute: PropTypes.bool,
+    showSecond: PropTypes.bool,
+    spinners: PropTypes.bool,
+    time: PropTypes.object
+};
```
Comment (greg-a-smith) [inline src/Time/Time.js]: Split these out so they can be shared by `TimePicker`.

File: src/_playground/App.scss [hunk 3]
```diff
@@ -198,3 +198,62 @@ $fd-fonts-path: '../../node_modules/fiori-fundamentals/dist/fonts/';
...
+                    }
+                }
+            }
+        }
+
+        .prop-required {
+            color: var(--fd-color-status-3);
+        }
+    }
+}
```
Comment (greg-a-smith) [inline src/_playground/App.scss]: Some custom override styling for the default Fiori table style.  It's mostly just reducing padding so the tables don't take up so much space.

File: src/_playground/documentation/Import/Import.js [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
     return (
-        <code>
-            import &#123; { arraySort(Object.keys(sourceModule)).join(', ') } &#125; from '{packageJson.name}';
-        </code>
+        <SyntaxHighlighter
+            customStyle={{padding: 0}}
+            language='javascript'
+            style={googlecode}>
+            {`import { ${Object.keys(sourceModule).sort().join(', ')} } from '${packageJson.name}';`}
+        </SyntaxHighlighter>
```
Comment (greg-a-smith) [inline src/_playground/documentation/Import/Import.js]: Small enhancement for the `import` statement so it is syntax highlighted.

File: src/_playground/documentation/Properties/Properties.js [hunk 5]
```diff
@@ -1,32 +1,192 @@
...
+import { Table } from '../../../Table/Table';
 
-export class Properties extends Component {
-    constructor(props) {
-        super(props);
-        this.state = { description: this.props.properties };
+export const Properties = ({ sourceModule }) => {
+    const componentNames = Object.keys(sourceModule).sort();
+
+    return (
```
Comment (bcullman) [inline src/_playground/documentation/Properties/Properties.js]: this is where the magic happens ;)

**Pull request merged**